# 📝 Chapter 1 Assignment: Probabilistic Models & Inference (SOLVED)

**Objective:** Test your understanding of Normal Distribution, PDF vs CDF, Z-scores, and Probability calculations.

**Context for Practical Questions (unless otherwise stated):**
Assume a dataset of student math scores where the distribution is **Normal** with:
*   **Mean ($\mu$) = 65**
*   **Standard Deviation ($\sigma$) = 9**

## 🧠 Part 1: Conceptual Understanding

Answer the following theoretical questions to solidify your understanding of the core concepts.

**1. What is the difference between PDF and CDF in a normal distribution?**

> **Answer:**
> *   **PDF (Probability Density Function):** Represents the height of the curve at a specific point $x$. It indicates the relative likelihood of the random variable being near that value. The area under the PDF curve over an interval gives the probability for that interval.
> *   **CDF (Cumulative Distribution Function):** Represents the area under the PDF curve from $-\infty$ to a specific point $x$. It gives the probability that the random variable $X$ will take a value less than or equal to $x$ ($P(X \le x)$).

**2. Why is the probability of a single value in a continuous distribution equal to zero?**

> **Answer:**
> In a continuous distribution, there are infinite possible values. Geometrically, probability is represented by the area under the curve. A single point is a line with a width of zero, so its area ($height \times 0$) is zero. We can only calculating probabilities for *ranges* of values.

**3. Explain why we standardize scores using z-scores before using the standard normal table.**

> **Answer:**
> Every normal distribution has a different mean ($\mu$) and standard deviation ($\sigma$). Standardizing converts these specific distributions into the **Standard Normal Distribution** ($Z \sim N(0, 1)$). This allows us to use a single table or function to calculate probabilities for *any* normal distribution.

**4. What does a z-score of −2 indicate in terms of the mean and standard deviation?**

> **Answer:**
> A z-score of -2 indicates that the raw score is exactly **2 standard deviations below the mean**.

**5. How is the area under the curve related to probability in a normal distribution?**

> **Answer:**
> The total area under the entire normal curve is equal to 1 (representing 100% probability). The area under the curve between two points $a$ and $b$ represents the probability that the random variable falls between those two values ($P(a \le X \le b)$).

**6. Why does the normal distribution integral not have a closed-form solution in elementary functions?**

> **Answer:**
> The integral of the Gaussian function ($e^{-x^2}$) cannot be computed using standard algebraic, trigonometric, or exponential functions (except over the full range $-\infty$ to $+\infty$). This is why we rely on numerical approximations (like Z-tables or computer algorithms like `scipy.stats`) to find probabilities.

**7. What does it mean if P(a ≤ X ≤ b) = 0.70? How would this appear on a normal distribution curve?**

> **Answer:**
> It means there is a 70% chance that a randomly selected value falls between $a$ and $b$. Visually, this would look like a shaded region under the curve between the vertical lines at $a$ and $b$ that occupies 70% of the total area.

**8. If the standard deviation decreases while the mean remains constant, how does the shape of the normal distribution change?**

> **Answer:**
> The curve becomes **narrower and taller**. The peak (at the mean) gets higher because the data is more clustered around the center, and the tails drop off more sharply.

## 📚 Part 2: Calculations & Z-Scores

**Given: $\mu = 65, \sigma = 9$**

Use Python or manual calculation to solve.

In [1]:
import scipy.stats as stats

mu = 65
sigma = 9

**1. Convert a raw score of 72 into a z-score.**

In [2]:
raw_score = 72
z_score = (raw_score - mu) / sigma
print(f"Z-score for {raw_score}: {z_score:.4f}")
# Calculation: (72 - 65) / 9 = 7 / 9 = 0.777...

Z-score for 72: 0.7778


**2. A student scored 50. How many standard deviations is this from the mean?**

In [3]:
raw_score = 50
z_score = (raw_score - mu) / sigma
print(f"Standard deviations from mean: {z_score:.4f}")
# Answer: It is approx 1.67 standard deviations BELOW the mean.

Standard deviations from mean: -1.6667


**3. If z = 1.25, what is the corresponding raw score?**

In [4]:
z = 1.25
raw_score = mu + (z * sigma)
print(f"Raw score for Z={z}: {raw_score}")
# Calculation: 65 + (1.25 * 9) = 65 + 11.25 = 76.25

Raw score for Z=1.25: 76.25


**4. Find P(X ≤ 70). (Hint: Use CDF)**

In [5]:
prob_less_70 = stats.norm.cdf(70, mu, sigma)
print(f"P(X <= 70): {prob_less_70:.4f}")

P(X <= 70): 0.7107


**5. Find P(X ≥ 80). (Hint: 1 - CDF)**

In [6]:
prob_greater_80 = 1 - stats.norm.cdf(80, mu, sigma)
# Alternatively: stats.norm.sf(80, mu, sigma) # sf = Survival Function (1-CDF)
print(f"P(X >= 80): {prob_greater_80:.4f}")

P(X >= 80): 0.0478


**6. Compute the probability that X lies between 60 and 75.**

In [7]:
prob_between = stats.norm.cdf(75, mu, sigma) - stats.norm.cdf(60, mu, sigma)
print(f"P(60 <= X <= 75): {prob_between:.4f}")

P(60 <= X <= 75): 0.5775


## 📊 Part 3: Word Problems

**Problem 1: Probability of Being Above Average**
The mean math score is 65, $\sigma$ = 9. What is the probability that a randomly selected student scores above 65?

In [8]:
# Theoretical: 65 is the Mean. The normal distribution is symmetric.
# Therefore, 50% of the scores are above the mean.
prob_above_mean = 1 - stats.norm.cdf(65, mu, sigma)
print(f"Probability > 65: {prob_above_mean:.2f}")

Probability > 65: 0.50


**Problem 2: Probability in a Range**
Find the probability that a student scores between 55 and 75.

In [9]:
prob = stats.norm.cdf(75, mu, sigma) - stats.norm.cdf(55, mu, sigma)
print(f"Probability 55-75: {prob:.4f}")

Probability 55-75: 0.7335


**Problem 3: Extreme Values**
What is the probability that a student scores below 50 or above 85?

In [10]:
p_below_50 = stats.norm.cdf(50, mu, sigma)
p_above_85 = 1 - stats.norm.cdf(85, mu, sigma)
total_prob = p_below_50 + p_above_85
print(f"P(<50) + P(>85) = {total_prob:.4f}")

P(<50) + P(>85) = 0.0609


**Problem 4: Comparing Two Intervals**
Compare the probability of scoring between 60 and 80 with the probability of scoring between 50 and 70. Which is higher and why?

In [11]:
p_60_80 = stats.norm.cdf(80, mu, sigma) - stats.norm.cdf(60, mu, sigma)
p_50_70 = stats.norm.cdf(70, mu, sigma) - stats.norm.cdf(50, mu, sigma)

print(f"P(60-80): {p_60_80:.4f}")
print(f"P(50-70): {p_50_70:.4f}")

diff_60 = 60 - mu # -5
diff_80 = 80 - mu # +15

diff_50 = 50 - mu # -15
diff_70 = 70 - mu # +5

print(f"\nDistance from mean (60-80): {diff_60} to {diff_80}")
print(f"Distance from mean (50-70): {diff_50} to {diff_70}")

P(60-80): 0.6630
P(50-70): 0.6630

Distance from mean (60-80): -5 to 15
Distance from mean (50-70): -15 to 5


**Conceptual Follow-up:**
Explain why the two ranges (60–80) and (50–70) in Problem 4 have the same probability, even though the widths are the same.

> **Answer:**
> The Normal Distribution is **symmetric** around the mean.
> *   Range A (60 to 80) covers deviations from $-5$ to $+15$.
> *   Range B (50 to 70) covers deviations from $-15$ to $+5$.
> 
> Because the curve is a mirror image, the area from -15 to -5 is exactly identical to the area from +5 to +15. Similarly, the area from -5 to +5 is shared/symmetric. Thus, the total areas are identical.

## 🔍 Part 4: Inverse Normal Problems & Advanced Reasoning

**1. Finding Percentiles**
If $X \sim N(65, 9^2)$, find the value of X such that 90% of students score below it (use z = 1.28 for 90%).

In [12]:
# Manual: X = mu + (z * sigma)
x_90 = 65 + (1.28 * 9)

# Precise Python way (PPF = Percent Point Function / Inverse CDF)
x_90_precise = stats.norm.ppf(0.90, mu, sigma)

print(f"90th Percentile (Manual): {x_90:.2f}")
print(f"90th Percentile (Precise): {x_90_precise:.2f}")

90th Percentile (Manual): 76.52
90th Percentile (Precise): 76.53


**2. Top Performers**
What score corresponds to the top 5% of students? (Use z = 1.645 for 95% cumulative probability.)

In [13]:
# Top 5% means 95% are below.
x_top5 = 65 + (1.645 * 9)
print(f"Score required for top 5%: {x_top5:.2f}")

Score required for top 5%: 79.81


**3. 80th Percentile**
A student’s score is at the 80th percentile. Estimate their score (use z ≈ 0.84).

In [14]:
x_80 = 65 + (0.84 * 9)
print(f"80th Percentile Score: {x_80:.2f}")

80th Percentile Score: 72.56


**4. Theory: Area Under Curve**
Why is the area under the normal curve always equal to 1?

> **Answer:**
> In probability theory, the sum of probabilities for all possible mutually exclusive outcomes must equal 1 (certainty). The integral from $-\infty$ to $+\infty$ of the PDF represents the sum of all possible probabilities for that random variable, so it must equal 1.

**5. Theory: Real-World Application**
Discuss the real-world importance of z-scores in comparing test results from different exams.

> **Answer:**
> Z-scores provide a standardized metric to compare performance across different scales. For example, if a student gets 600 on the SAT (scaled 400-1600) and 25 on the ACT (scaled 1-36), looking at raw scores tells us nothing. By converting both to Z-scores, we can see exactly how well the student performed relative to their peers in each specific test. If SAT Z=1.5 and ACT Z=1.2, they performed relatively better on the SAT.

**6. Theory: Standard Normal Preference**
Why do statisticians prefer using the standard normal distribution instead of working with raw values?

> **Answer:**
> It simplifies calculation and communication. Instead of creating a new probability table for every single experiment's specific mean and standard deviation, we typically translate everything to $Z \sim N(0,1)$, solve the problem in "Z-space", and then translate back if necessary. It makes the math universal.